vermouth/rcsu/contact_map.py

```diff
@@ -12,19 +12,672 @@
+        'CD2':      {'vrad': 1.76, 'atype': 5},
+        'CE1':      {'vrad': 1.76, 'atype': 5},
+        'CE2':      {'vrad': 1.76, 'atype': 5},
+        'CZ':       {'vrad': 1.61, 'atype': 5},
+        'OH':       {'vrad': 1.46, 'atype': 1},
+        'OXT':      {'vrad': 1.42, 'atype': 2},
+        'default':  {'vrad': 0.00, 'atype': 0},
+    },
+    "VAL": {
+        'N':        {'vrad': 1.64, 'atype': 3},
+        'CA':       {'vrad': 1.88, 'atype': 7},
+        'C':        {'vrad': 1.61, 'atype': 6},
+        'O':        {'vrad': 1.42, 'atype': 2},
+        'CB':       {'vrad': 1.88, 'atype': 4},
+        'CG1':      {'vrad': 1.88, 'atype': 4},
+        'CG2':      {'vrad': 1.88, 'atype': 4},
+        'OXT':      {'vrad': 1.42, 'atype': 2},
+        'default':  {'vrad': 0.00, 'atype': 0},
+    }
+}
+
+
+def _get_vdw_radius(resname, atomname):
+    """
+    get the vdw radius of an atom indexed internally within a serially numbered residue
+    """
+    res_vdw = PROTEIN_MAP[resname]
+    try:
+        atom_vdw = res_vdw[atomname]
+    except KeyError:
+        atom_vdw = res_vdw['default']
+    return atom_vdw['vrad']
+
+
+def _get_atype(resname, atomname):
+    """
+    get the vdw radius of an atom indexed internally within a serially numbered residue
+    """
+    res_vdw = PROTEIN_MAP[resname]
+    try:
+        atom_vdw = res_vdw[atomname]
+    except KeyError:
+        atom_vdw = res_vdw['default']
+    return atom_vdw['atype']
+
+
+def make_surface(position, fiba, fibb, vrad):
+    """
+    Generate points on a sphere using Fibonacci points
+
+    position: np.array
+        shape (3,) array of an atomic position to build a sphere around
+    fiba: int. n-1 fibonacci number to build number of points on sphere
+    fibb: int. n fibonacci number to build number of points on sphere
+    vrad: float. VdW radius of the input atom to build a sphere around.
+
+    position: centre of sphere
+    """
+
+    x, y, z = position
+
+    k = np.arange(fibb)
+    phi_aux = (np.arange(1, fibb+1) * fiba) % fibb
+    phi_aux[phi_aux == 0] = fibb
+    theta = np.arccos(1.0 - 2.0 * k / fibb)
+    phi = 2.0 * np.pi * phi_aux / fibb
+    surface_x = x + vrad * np.sin(theta) * np.cos(phi)
+    surface_y = y + vrad * np.sin(theta) * np.sin(phi)
+    surface_z = z + vrad * np.cos(theta)
+    surface = np.stack((surface_x, surface_y, surface_z), axis=-1)
+
+    return surface
+
+
+def atom2res(arrin, nresidues, atom_map, norm=False):
+    '''
+    take an array with atom level data and sum the entries over within the residue
+    '''
+
+    out = np.zeros((nresidues, nresidues))
+    for res_idx, res_jdx in product(np.arange(nresidues), np.arange(nresidues)):
+        atom_idxs = np.array(atom_map[res_idx])
+        atom_jdxs = np.array(atom_map[res_jdx])
+        value = arrin[atom_idxs,
+        atom_jdxs[:, np.newaxis]].sum()
+        out[res_idx, res_jdx] = value
+
+    if norm:
+        out[out > 0] = 1
+
+    return out
```

Comment:
Don't make np arrays where you don't need them. The atom_map is a dict of resid: tuple of node indices?
```suggestion
    nresidues = len(atom_map)  # Right?
    out = np.zeros((nresidues, nresidues))
    for res_idx, res_jdx in product(atom_map.keys(), atom_map.keys()):
        atom_idxs = atom_map[res_idx]
        atom_jdxs = atom_map[res_jdx]
        value = arrin[atom_idxs, atom_jdxs].sum()
        out[res_idx, res_jdx] = value

    if norm:
        out[out > 0] = 1

    return out
```